include/seqan3/argument_parser/detail/format_base.hpp

```diff
@@ -464,6 +464,23 @@ class format_help_base : public format_base
     std::vector<std::string> command_names{};
     //!\brief Whether to show advanced options or not.
     bool show_advanced_options{true};
+
+private:
+    /*!\brief Checks whether the information annotated with `spec` should be added to the help page.
+     * \param[in] spec The option specification.
+     * \returns `true` if the information annotated with `spec` should be added to the help page, `false` otherwise.
+     *
+     * \details
+     *
+     * If `spec` equals `seqan3::option_spec::HIDDEN`, the information is never added to the help page.
+     * If `spec` equals `seqan3::option_spec::ADVANCED`, the information is only added to the help page if
+     * the advanced help page has been queried on the command line (`show_advanced_options == true`).
+     */
+    bool information_should_be_added(option_spec const spec)
```

Comment:
ah ok. I had such a design in mind:
```cpp
void store_help_page_element_if(option_spec const spec, std::function<void()> print_callback)
{
   if (!(spec & option_spec::HIDDEN) && (!(spec & option_spec::ADVANCED) || show_advanced_options))
       parser_set_up_calls.push_back(std::move(print_callback));
}
```

Then you can simply call it in a single line:
```cpp
store_help_page_element_if(spec, [this, key, desc] () { derived_t().print_list_item(key, desc); });
```